src/main/java/com/google/firebase/auth/UserRecord.java

```diff
@@ -569,7 +604,24 @@ UpdateRequest setValidSince(long epochSeconds) {
       }
 
       if (copy.containsKey("phoneNumber") && copy.get("phoneNumber") == null) {
-        copy.put("deleteProvider", ImmutableList.of("phone"));
+        Object deleteProvider = copy.get("deleteProvider");
+        if (deleteProvider != null) {
+          if (!(deleteProvider instanceof Iterable<?>)) {
+            throw new IllegalStateException("Contents of deleteProvider was not iterable");
```

Comment:
I'd just forego the explicit check and let the cast error pop up. This case should never occur since the writes to the properties map is managed by our code.